### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by wmontgomery4**
This includes the following
- Fix memory "leak" by removing old samples from agent. Pretty sure this is okay to do since we never need the ones from previous iterations
- Fix `sample` method for `agent_box2d`
- Fix `stepadjust` method in `algorithm_mdgps` (was using `arange` instead of `empty`, leading to integer arrays)
- Fix `mjc_mdgps_example` (I broke this on a previous commit when changing the `stepadjust` method, sorry!)

### CODE REVIEW

File: python/gps/gps_main.py [hunk 1]
```diff
@@ -69,10 +69,14 @@ def run(self, itr_load=None):
                 self.agent.get_samples(cond, -self._hyperparams['num_samples'])
                 for cond in self._train_idx
             ]
+
             self._take_iteration(itr, traj_sample_lists)
             pol_sample_lists = self._take_policy_samples()
             self._log_data(itr, traj_sample_lists, pol_sample_lists)
 
+            # Clear agent samples to prevent memory leak.
+            self.agent.clear_samples()
```
Comment (cbfinn) [inline python/gps/gps_main.py]: I think this can safely go right before _take_iteration().

File: python/gps/gps_main.py [hunk 2]
```diff
@@ -69,10 +69,14 @@ def run(self, itr_load=None):
                 self.agent.get_samples(cond, -self._hyperparams['num_samples'])
                 for cond in self._train_idx
             ]
+
             self._take_iteration(itr, traj_sample_lists)
             pol_sample_lists = self._take_policy_samples()
             self._log_data(itr, traj_sample_lists, pol_sample_lists)
 
+            # Clear agent samples to prevent memory leak.
```
Comment (cbfinn) [inline python/gps/gps_main.py]: nitpic: maybe just "Clear agent samples". It's not exactly a memory leak, more like excessive memory usage.

**Review by cbfinn — CHANGES_REQUESTED** [review]
Other than the two minor comments, LGTM

**Comment by wmontgomery4** [comment]
Okay, made those changes and squashed the commit, and mjc_mdgps_example is working. Merging.

**Pull request merged**